independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InjectionPointsTransformer.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.arc.processor;
+
+import java.lang.annotation.Annotation;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+
+/**
+ * Allows a build-time extension to alter qualifiers on an injection point.
+ * <p>
+ *
+ * @author Matej Novotny
+ */
+public interface InjectionPointsTransformer extends BuildExtension {
+
+    /**
+     * Type of injection point to which this transformer applies.
+     *
+     * @param typeOfInjectionPoint the declared type of the injection point
+     */
+    //TODO compared to PIP in CDI, we miss info on declaring class, can that be done in Jandex? Else this is pretty broad...
+    boolean appliesTo(DotName typeOfInjectionPoint);
```

Comment:
We use this to optimize/filter out what does the transformer apply to, I'm not sure what's best in Jandex from user perspective. `DotName` versus `Type`?

For the record, in CDI you have the type of the bean and the class in which it is declared. Oftentimes you want all classes (using wildcard), hence I used only the "type" part.

Reply:
I'm quite sure we should use `Type` so that it's possible to filter parameterized types. E.g. for `List<String>` the `DotName` would be just `java.util.List`.

Reply:
Good point, did that.